apps/prairielearn/src/models/course.ts

```diff
@@ -230,3 +231,25 @@ export async function updateCourseSharingName({ course_id, sharing_name }): Prom
     sharing_name,
   });
 }
+
+/**
+ * Look up courses by sharing names (may return null if non-existent)
+ */
+export async function findCoursesBySharingNames(
```

Comment:
nit: I don't typically see a lot of usage of `Map` in our codebase. Consider using a plain `Record`.